pkg/utils/ssh.go

```diff
@@ -18,10 +19,67 @@ import (
 // GetSSHConnectionString returns the SSH connection string for the given public IP and certificate file path.
 func GetSSHConnectionString(publicIP, certFilePath string) string {
 	if certFilePath != "" {
-		return fmt.Sprintf("ssh %s %s@%s -i %s", constants.AnsibleSSHShellParams, constants.AnsibleSSHUser, publicIP, certFilePath)
-	} else {
-		return fmt.Sprintf("ssh %s %s@%s", constants.AnsibleSSHUseAgentParams, constants.AnsibleSSHUser, publicIP)
+		certFilePath = fmt.Sprintf("-i %s", certFilePath)
 	}
+	return fmt.Sprintf("ssh %s %s@%s %s", constants.AnsibleSSHShellParams, constants.AnsibleSSHUser, publicIP, certFilePath)
+}
+
+// GetSCPCommandString returns the SCP command string for the given source and destination paths.
+func GetSCPCommandString(certFilePath string, sourceIP, sourcePath string, destIP, destPath string, recursive, withCompression bool) (string, error) {
+	scpParams := constants.AnsibleSSHShellParams + " -B -o LogLevel=Error"
+	if sourceIP == "" && destIP == "" {
+		return "", fmt.Errorf("source or destination should be remote")
+	}
+	if sourceIP == "" && destPath == "" {
+		return "", fmt.Errorf("source and destination path is required")
+	}
+	// end of checks
+	if recursive {
+		scpParams += " -r"
+	}
+	if withCompression {
+		scpParams += " -C"
+	}
+	if certFilePath != "" {
+		scpParams += fmt.Sprintf(" -i %s", certFilePath)
+	}
+	if sourceIP != "" && destIP != "" {
+		scpParams += " -3"
+	}
+	if sourceIP != "" {
+		sourcePath = fmt.Sprintf("%s@%s:%s", constants.AnsibleSSHUser, sourceIP, sourcePath)
```

Comment:
can you use a function for this? probably overload the function you have in this file

Reply:
moved

Reply:
done